### PR TITLE
The Daily Planner by Kartika

### PR DESCRIPTION
**Pull request by katirum**
Link to my deployed project:
https://thedailyplanner.netlify.app/

### CODE REVIEW

File: code/public/index.html [hunk 1]
```diff
@@ -13,7 +13,32 @@
...
+    <meta property="twitter:url" content="https://thedailyplanner.netlify.app/" />
+    <meta name="twitter:title" content="Daily Planner" />
+    <meta name="twitter:description" content="Go through your day sophisticatedly with our daily planner app." />
+    <meta name="twitter:image" content="https://thedailyplanner.netlify.app/assets/ogimage.jpg" />
+
+    <!-- Icon on the tab title -->
+    <link rel="apple-touch-icon" sizes="180x180" href="/assets/apple-touch-icon.png">
+    <link rel="icon" type="image/png" sizes="32x32" href="/assets/favicon-32x32.png">
+    <link rel="icon" type="image/png" sizes="16x16" href="/assets/favicon-16x16.png">
+    <link rel="manifest" href="/site.webmanifest">
```
Comment (MariaWellander) [inline code/public/index.html:36]: Looks really professional with these meta tags and icon for the tab title!

File: code/package-lock.json [hunk 2]
```diff
@@ -9,14 +9,19 @@
...
         "eslint-plugin-import": "^2.26.0",
         "eslint-plugin-jsx-a11y": "^6.6.1",
         "eslint-plugin-react": "^7.30.1",
         "eslint-plugin-react-hooks": "^4.6.0",
         "react": "^18.2.0",
-        "react-dom": "^18.2.0"
+        "react-dom": "^18.2.0",
+        "react-redux": "^8.0.4",
+        "styled-components": "^5.3.6",
+        "uniqid": "^5.4.0"
```
Comment (MariaWellander) [inline code/package-lock.json:24]: Cool that you are using this!

File: code/src/components/AddTask.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    dispatch(tasks.actions.addItem(newTask))
+
+    setvalue('')
+  }
+
+  return (
+    <Form onSubmit={onFormSubmit}>
+      <Description
+        placeholder="What do you want to do today?"
```
Comment (MariaWellander) [inline code/src/components/AddTask.js:32]: Nicve with a placeholder text!

File: code/src/components/Greeting.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  const date = new Date();
+  const hours = date.getHours();
+
+  if (hours < 10) {
+    timeOfDay = 'morning';
+  } else if (hours >= 11 && hours < 17) {
+    timeOfDay = 'afternoon';
+  } else {
+    timeOfDay = 'evening';
+  }
```
Comment (MariaWellander) [inline code/src/components/Greeting.js:15]: Really fun that you used this in this project too - looks very "skilled" (and so are you!!)!

File: code/src/components/Header.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    margin-top: 10px;
+    font-weight: 400;
+  }
+
+  .line{
+    height: 1px;
+    margin-top: 20px;
+    flex: 1;
+    background-color: var(--quaternary);
+  }
```
Comment (MariaWellander) [inline code/src/components/Header.js:62]: Your header looks very professional and I like the design a lot :) Kinda a news paper touch to it - wonderful!

File: code/src/components/TaskCounter.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  const completeTodo = useSelector(
+    (store) => store.tasks.items.filter((item) => item.isDone)
+  );
+
+  return (
+    <Counter>
+          &quot;{completeTodo.length} out of {todoCounter} task is done&quot;
+    </Counter>
+  );
+};
```
Comment (MariaWellander) [inline code/src/components/TaskCounter.js:16]: Very sleek and tidy!

File: code/package-lock.json [hunk 7]
```diff
@@ -9,14 +9,19 @@
       "version": "1.0.0",
       "dependencies": {
         "@babel/eslint-parser": "^7.18.9",
+        "@reduxjs/toolkit": "^1.8.6",
+        "date-fns": "^2.29.3",
```
Comment (MariaWellander) [inline code/package-lock.json:13]: Good solution for the dates :)

File: code/src/components/TaskList.js [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+    place-content: center;
+  }
+
+  &:checked + label{
+    text-decoration: 4px line-through var(--primary);
+  }
+  
+  &:checked::before{
+    transform:scale(1)
+  }
```
Comment (MariaWellander) [inline code/src/components/TaskList.js:171]: Ah, the custom checkboxes - fantastic that you coded that!

File: code/src/components/TaskList.js [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+              type="checkbox"
+              checked={newTodo.isDone}
+              onChange={() => onToDoToggle(newTodo.id)}
+              onClick={() => toggleLineThrough(this)} />
+            <label
+              htmlFor="a"
+              id="task">
+              {newTodo.name}
+            </label>
+            <p>{formatRelative(new Date(), new Date())}</p>
```
Comment (MariaWellander) [inline code/src/components/TaskList.js:45]: Extremely nice!

File: code/src/components/TodaysDate.js [hunk 10]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { format } from 'date-fns';
+import styled from 'styled-components';
+
+const TodaysDate = () => {
+  return (
+
+    <NowDate>Today is {format(new Date(), 'MMMM do')}&quot; </NowDate>
```
Comment (MariaWellander) [inline code/src/components/TodaysDate.js:8]: I really like this in the header :)

File: code/src/styledcomponents/GlobalStyles.js [hunk 11]
```diff
@@ -0,0 +1,99 @@
+import styled, { createGlobalStyle } from 'styled-components';
```
Comment (MariaWellander) [inline code/src/styledcomponents/GlobalStyles.js:1]: Nice with global styles - and styled components! Styled components is a red requirement :O

**Review by MariaWellander — COMMENTED** [review]
### Am I able to understand the code easily? Does the code have lots of duplicates? Could it be tidied up? ###

I'm absolutely able to understand the code easily - good job with that. It looks really neat and clean.

### Does it work or does anything look broken? ###

It work's and nothing seems broken - well done!

### Does it follow the general and blue requirements? ###


Yes! It even meets up the requirements for the red level - impressive!

### To sum it up ###

Wonderful work, Tika! You should be very proud of yourself :D I really like the design and the news paper touch to it! High five!
